src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -624,6 +624,39 @@ int D4NFilterBucket::list(const DoutPrefixProvider* dpp, ListParams& params, int
   return 0;
 }
 
+int D4NFilterBucket::remove(const DoutPrefixProvider* dpp,
+			    bool delete_children,
+			    optional_yield y)
+{
+  ListParams params;
+  ListResults results;
+  cache_request = true; // though not a cache request, this helps limit the scope of the list call to what is in the cache
+  int ret = list(dpp, params, 1000, results, y);
```

Comment:
This only returns accurate results in the case where the D4N write cache is active. Otherwise, the op fails

Reply:
Here you are just listing 1000 objects in the bucket. what if there are more? You need to check for is_truncated flag and get more elements. Check how it is being done in Rados.

Reply:
The bucket list operation fetches data from cache only when write back cache is active for 2 reasons 1. in case the write back is not active, then the objects reside in the backend also 2. the versions are fetched from the backend in case when write-back is not active. You can use the cache_request flag to delete entries in the cache and the call to backend will take care of removing the entries there anyway.